### PR TITLE
 Add ability to start ExDTLS with custom key/cert

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: lib/ex_dtls.ex [hunk 1]
```diff
@@ -60,6 +66,36 @@ defmodule ExDTLS do
...
+
+  @doc """
+  Gets current private key.
+
+  Returns key specific representation in binary format.
+  """
+  @spec get_pkey(pid :: pid()) :: pkey :: binary()
+  def get_pkey(pid) do
+    GenServer.call(pid, :get_pkey)
+  end
```
Comment (mat-hek) [inline lib/ex_dtls.ex:87]: I'm wondering if this could automatically start and kill the CNode

File: lib/ex_dtls.ex [hunk 2]
```diff
@@ -60,6 +66,36 @@ defmodule ExDTLS do
...
+
+  @doc """
+  Gets current certificate.
+
+  Returns DER representation in binary format.
+  """
+  @spec get_cert(pid :: pid()) :: cert :: binary()
+  def get_cert(pid) do
+    GenServer.call(pid, :get_cert)
+  end
```
Comment (mat-hek) [inline lib/ex_dtls.ex:97]: as above

File: c_src/ex_dtls/native.c [hunk 3]
```diff
@@ -9,13 +9,57 @@
...
+    res_term = unifex_raise(env, "Cannot generate cert");
+    goto exit;
+  }
+
+  res_term = do_init(env, client_mode, dtls_srtp, pkey, x509);
+exit:
+  return res_term;
+}
+
+UNIFEX_TERM init_var(UnifexEnv *env, int client_mode, int dtls_srtp,
```
[the remark this answers is not in the document]
Comment (mickel8) [reply c_src/ex_dtls/native.c:39]: The idea was we initialize ExDTLS using some extra/optional variables. Maybe `init_from_key_cert` instead?

File: lib/ex_dtls.ex (hunk 1 above)
Comment (mickel8) [reply lib/ex_dtls.ex:87]: I am not sure if adding additional logic here is a good idea. `ExDTLS` is written as CNode so it's reasonable to start CNode at first. Also we might want to get pkey/cert of an existing ExDTLS instance e.g. for decrypting packets in Wireshark (in the future).

File: c_src/ex_dtls/native.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (mat-hek) [reply c_src/ex_dtls/native.c:39]: sounds better

File: lib/ex_dtls.ex (hunk 1 above)
Comment (mat-hek) [reply lib/ex_dtls.ex:87]: Ok, so I'd add some `stop` function so that one doesn't have to kill the process manually

File: lib/ex_dtls.ex [hunk 4]
```diff
@@ -242,6 +250,12 @@ defmodule ExDTLS do
   def handle_call(:get_cert, _from, %State{cnode: cnode} = state),
     do: {:reply, Unifex.CNode.call(cnode, :get_cert), state}
 
+  @doc false
+  @impl true
```
Comment (mat-hek) [inline lib/ex_dtls.ex:254]: `impl` automatically adds `doc false`

**Review by mat-hek — APPROVED** [review]

**Pull request merged**